app/components/avo/index/resource_controls_component.rb

```diff
@@ -65,8 +65,8 @@ def singular_resource_name
   end
 
   def parent_resource
+    return @parent_resource unless @parent_resource.blank?
     return nil if @parent_model.blank?
-
```

Comment:
It's a common and good practice to insert a newline after a guard clause to increase readability